fastlane/fastlane
# [spaceship] add app preview

# Motivation and Context
Fixes https://github.com/fastlane/fastlane/pull/16640#issuecomment-648289796

### Description
- Added `AppPreviewSet`
- Added `AppPreview`

### Testing Steps

```rb

  lane :preview do                                                                                                                                                      
    fastlane_require 'spaceship'                                                                                 
                                                                                                                 
    Spaceship::Tunes.login("email")                                                                   
    Spaceship::Tunes.select_team(team_name: "team")                                                        
                                                                                                                 
    app = Spaceship::ConnectAPI::App.find("com.bundle.id")                                           
    version = app.get_edit_app_store_version(platform: "IOS")                                                    
                                                                                                                 
    localizations = version.get_app_store_version_localizations                                                  
    localizations.each do |localization|                                                                         
      puts "Locale: #{localization.locale}"                                                                      
                                                                                                                 
      preview_sets = localization.get_app_preview_sets                                                           
      preview_sets.each do |preview_set|                                                                         
        puts "\t#{preview_set.preview_type} #{preview_set.app_previews.size}"                                    
      end                                                                                                        
                                                                                                                 
      iphone_65 = preview_sets.find do |set|                                                                     
        set.preview_type == Spaceship::ConnectAPI::AppPreviewSet::PreviewType::IPHONE_65                         
      end                                                                                                        
      if iphone_65.nil?                                                                                          
        iphone_65 = localization.create_app_preview_set(attributes: {                                            
          previewType: Spaceship::ConnectAPI::AppPreviewSet::PreviewType::IPHONE_65                              
        })                                                                                                       
      end                                                                                                        
                                                                                                                 
      if !iphone_65.app_previews.empty? && UI.confirm("Delete all previews?")                                    
        iphone_65.app_previews.each do |app_preview|                                                             
          UI.message("Deleting #{app_preview.id}")                                                               
          app_preview.delete!                                                                                    
          UI.message("Deleted #{app_preview.id}")                                                                
        end                                                                                                                
      end                                                    
                                                                                                                                 
      if UI.confirm("Upload new preview?")                      
        path = File.absolute_path("previews/iphone65-en-US-1-portrait.mp4")                                                       
        iphone_65.upload_preview(path: path)                       
      end                        
    end
  end
```
